syft/pkg/cataloger/javascript/cataloger.go

```diff
@@ -20,3 +20,13 @@ func NewLockCataloger() *generic.Cataloger {
 		WithParserByGlobs(parseYarnLock, "**/yarn.lock").
 		WithParserByGlobs(parsePnpmLock, "**/pnpm-lock.yaml")
 }
+
+// NewJavaScriptCataloger returns a new JavaScript cataloger object based on detection
+// of npm based packages and lock files to provide a complete dependency graph of the
+// packages.
+func NewJavaScriptCataloger() *generic.GroupedCataloger {
```

Comment:
I think this would need to be reconciled with the `NewLockCataloger()` above. At surface glance this cataloger would replace the lock cataloger.

Reply:
yep, that's true, was looking for guidance here on how to deprecate an existing cataloger on favor of a new cataloger and what the process would be for that

Reply:
I've just picked this back up for re-review today. I'm going to try and help get the PR rebased and working again, so I'll be pushing some commits. I'll get back to this question while I'm trying to square it with the current state on main.